LSPS1/README.md

```diff
@@ -122,6 +119,8 @@ The client MUST call `lsps1.info` first.
 
 Every `min/max` options pair MUST ensure that `min <= max`.
 
+An LSP, and newer versions of this LSPS, MUST NOT add new fields to the `options` dictionary.
+Future versions of this LSPS, and extensions to this LSPS, MAY add new entire fields in the result, such as adding a new `v2_options` dictionary, which only apply if the client speifically requests for using those options.
```

Comment:
I don't know if I understand this correctly so let me know if I made any logical mistakes:

- In LSPS1, we are not allowed to add an optional option. What is the reason behind this?
- In future LSPSx, we should be able to change whatever we want. Why are we not allowed to completely redo `options`?

Reply:
>In LSPS1, we are not allowed to add an optional option. What is the reason behind this?

Because the additional option may imply that the characteristics of the channel may be different from what a pre-this-new-option client expects.  This is why a future revision of LSPS1 that wants to add a new option needs to put those in a new `v2_options` field, and which the client must specifically opt-into so that the LSP knows that the client understands the new options.

>In future LSPSx, we should be able to change whatever we wan

True.  That is why the text says "this LSPS" and not "any future LSPS".

Reply:
> True. That is why the text says "this LSPS" and not "any future LSPS".

I have trouble understanding what is "Future versions of this LSPS". It is clear now because you elaborated on it but it is hard to get just from reading the PR.
Problem: From a language POV, even a non-breaking change is a new version.
Can we define standardize language around this so it gets very clear what the spec means?
Borrowing definitions from [SemVer](https://semver.org/):
- "Major" version when you make incompatible API changes.
- "Minor" version when you add functionality in a backward compatible manner.

What do you think?

>Because the additional option may imply that the characteristics of the channel may be different from what a pre-this-new-option client expects. This is why a future revision of LSPS1 that wants to add a new option needs to put those in a new v2_options field, and which the client must specifically opt-into so that the LSP knows that the client understands the new options.

Is this necessary? Is there no way to add an additional option if the new LSPS1 guarantees to use backward compatible default options from the previous (minor) version?

Reply:
> Is there no way to add an additional option if the new LSPS1 guarantees to use backward compatible default options from the previous (minor) version?

What happens if the LSP gives a non-default option that is defined in a later version?

Reply:
Resolved in 2023-11-16: remove the restriction on `options`.